### PR TITLE
Fix endgame component

### PR DESCRIPTION
**Pull request by LoserAntbear**
- Upgrade ts version
- Reconfigure ts and linter
- Move <Finish> component to own namespace
- Add interfaces
- Add global Callback type
- Add locale and finish messages
- Add gamemodes constant
- Move <Button> component to ts
- Move <Finish> component to ts
- Move actions to ts

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react';
 import { Route } from 'react-router-dom';
 import './App.css';
 
-import Finish from './components/Finish';
+import Finish from './components/finish/';
```
Comment (basecoplan) [inline src/App.tsx]: Why did you rename finish component to lowercase?

File: src/components/finish/Finish.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Finish extends Component {
   render() {
     return (
       <section className="Finish">
-        <h1>{this.props.userWin ? 'THE VICTORY IS YOURS' : 'WASTED'}</h1>
+        <h1> {this.props.userWin ? 'THE VICTORY IS YOURS' : 'WASTED'} </h1>{' '}
```
Comment (basecoplan) [inline src/components/finish/Finish.js]: Redundant spaces and {' '}

File: src/components/button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import './style.css';
+
+type Props = {
+  title: string,
+} & Partial<Readonly<typeof defaultProps>>;
```
Comment (basecoplan) [inline src/components/button/index.tsx]: Is it right way of doing this?

File: src/actions/index.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+const restart = actions.restart;
+
+// TODO: define disathch and state-related interfaces
+const isWin = (state: any) => {
+  return state.game.user.win || state.game.ai.win;
+};
+
+const check = (value: any) => {
+  return (dispatch: any, getState: Callback) => {
```
Comment (basecoplan) [inline src/actions/index.ts]: getState it not a 'callback':)
Why didn't you specify state type return lamda?

File: src/interfaces/IUser.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// TODO: Define this interfaces
+export interface IUser {
+    stub: string;
```
Comment (basecoplan) [inline src/interfaces/IUser.ts]: Stub?
Is it for creating an interface?

File: src/components/finish/Finish.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import React, { Component } from 'react';
```
Comment (basecoplan) [inline src/components/finish/Finish.js]: I think this file is need anymore

Comment (basecoplan) [reply src/components/finish/Finish.js]: Delete it, please

File: src/interfaces/ILoadedState.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { IUser } from "./IUser";
+import { IAi } from "./IAI";
+import { GAME_MODES } from "../constants/gamemodes";
+
+export interface ILoadedState {
```
Comment (basecoplan) [inline src/interfaces/ILoadedState.ts]: ILoadedState 
Why did you call it 'Loaded'?

File: src/actions/actions.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import LocalStorage from '../localStorage/LocalStorage';
```
Comment (basecoplan) [inline src/actions/actions.ts]: Where is action.js removing?

File: src/actions/index.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import { push } from 'react-router-redux';
```
Comment (basecoplan) [inline src/actions/index.ts]: Where is index.js removing?

File: src/components/button/index.tsx (hunk 3 above)
Comment (LoserAntbear) [reply src/components/button/index.tsx]: It's completely right, what's wrong with this?

File: src/interfaces/IUser.ts (hunk 5 above)
Comment (LoserAntbear) [reply src/interfaces/IUser.ts]: It's literally a stub, 'cause you cannot leave empty Interface

File: src/interfaces/ILoadedState.ts (hunk 7 above)
Comment (LoserAntbear) [reply src/interfaces/ILoadedState.ts]: 'Cause it's loaded state?

Comment (basecoplan) [reply src/interfaces/ILoadedState.ts]: oh, I see, It's form local storage.

File: src/components/button/index.tsx (hunk 3 above)
Comment (basecoplan) [reply src/components/button/index.tsx]: Ok:)
